src/operations/exp.rs

```diff
@@ -0,0 +1,130 @@
+// Copyright 2015-2020 Aerospike, Inc.
+//
+// Portions may be licensed to Aerospike, Inc. under one or more contributor
+// license agreements.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+//! Expression Operations.
+//! This functions allow users to run `FilterExpressions` as Operate commands.
+
+use crate::commands::buffer::Buffer;
+use crate::errors::Result;
+use crate::expressions::FilterExpression;
+use crate::msgpack::encoder::{pack_array_begin, pack_integer};
+use crate::operations::{Operation, OperationBin, OperationData, OperationType};
+use crate::ParticleType;
+
+/// Expression write Flags
+pub enum ExpWriteFlags {
+    /// Default. Allow create or update.
+    Default = 0,
+    /// If bin does not exist, a new bin will be created.
+    /// If bin exists, the operation will be denied.
+    /// If bin exists, fail with Bin Exists
+    CreateOnly = 1,
+    /// If bin exists, the bin will be overwritten.
+    /// If bin does not exist, the operation will be denied.
+    /// If bin does not exist, fail with Bin Not Found
+    UpdateOnly = 2,
+    /// If expression results in nil value, then delete the bin.
+    /// Otherwise, return OP Not Applicable when NoFail is not set
+    AllowDelete = 3,
```

Comment:
This should be 4. Because I am also prone to this particular mistake, I prefer to write flags this way (YMMV):

```rust
// Assuming this also works in rust.
Default = 0,
CreateOnly = 1 << 0,
UpdateOnly = 1 << 1,
AllowDelete = 1 << 2,
PolicyNoFail = 1 << 3,
EvalNoFail = 1 << 4,
```

Reply:
Thanks for pointing out! Fixed.